File: src/utils/logging.js
// Logging utility functions
export function logInfo(logger, message) {
  logger.info(message);
}

export function logError(logger, error, context = '') {
  logger.error(`${context}${context ? ': ' : ''}${error}`);
}

export function logDebug(logger, message) {
  logger.debug(message);
}

export function logWarning(logger, message) {
  logger.warn(message);
}

File: src/utils/logging.go
package utils

import (
	"log"
)

// LogInfo logs an informational message
func LogInfo(message string) {
	log.Printf("[INFO] %s", message)
}

// LogError logs an error message with optional context
func LogError(err error, context string) {
	if context != "" {
		log.Printf("[ERROR] %s: %v", context, err)
	} else {
		log.Printf("[ERROR] %v", err)
	}
}

// LogDebug logs a debug message
func LogDebug(message string) {
	log.Printf("[DEBUG] %s", message)
}

// LogWarning logs a warning message
func LogWarning(message string) {
	log.Printf("[WARNING] %s", message)
}
